### DISCUSSION TITLE
ReactPaginate for unknown pageCount

### DISCUSSION THREAD

**Discussion by mariusseiceanu** [AdeleD/react-paginate#381] [Q&A] [answered]
Hi,
ReactPaginate is a very useful component and I would like to use it for paginating a table with an unknown number of results.
The pageCount is not known from the beginning and the user can click Next to show the next page - he can do this until the backend returns "no more results". Navigating back would be easy because we know how many previous pages we have.

I there a way to use ReactPaginate for such a usecase? 

Thanks,
Marius.

**Comment by carlos-dubon** [comment, marked as answer]
The backend is responsible for sending how many pages it has available for this specific use case. This is a must with API's that involve pagination.

If for some reason it doesn't send the amount of pages you could write a function to fetch all available pages and then get the amount of pages from it. This is not idealy for obvious reasons.. That's why the API needs to return the amount of pages.

**Comment by morosanmihail** [comment]
I've had to tackle this just today and my approach was to set page count to the (current page + 1) if the current page has a full page size of results. If using pagination using tokens, you can check for a next page token instead and set to (current +1) if there is, (current) if there isn't.

**Comment by MichaellIbarra** [comment]
like
